### DISCUSSION TITLE
Simu5g 1.2.2 build error : Undefined symbol

### DISCUSSION THREAD

**Discussion by shashini-w** [omnetpp/omnetpp#988] [Q&A] [answered]
I am trying to build simu5g 1.2.2 on windows 10 with INET 4.5 on OMNeT 6.0.1. I installed both INET and simu5g from the ONNET IDE itself. INET compiled without any issue but when I try to build simu5g, I am getting the following error about undefined symbols. 

I checked the mentioned files and everything is defined in INET and so I don't understand why it isn't possible for it to find these files. Has anyone faced this issue before and know how to solve it or if anyone has any hints on this I highly appreciate it.

```
ld.lld: error: undefined symbol: inet::CodeFragment::CodeFragment(void (*)(), omnetpp::SimulationLifecycleEventType)
referenced by ../out/gcc-release/src/apps/mec/DeviceApp/DeviceAppMessages/Serializers/DeviceAppPacketSerializer.o:(.text.startup)
referenced by ../out/gcc-release/src/apps/mec/DeviceApp/DeviceAppMessages/Serializers/DeviceAppPacketSerializer.o:(.text.startup)
referenced by ../out/gcc-release/src/apps/mec/DeviceApp/DeviceAppMessages/Serializers/DeviceAppPacketSerializer.o:(.text.startup)
referenced 14 more times

ld.lld: error: undefined symbol: inet::OperationalMixin<omnetpp::cSimpleModule>::isUp() const
referenced by ../out/gcc-release/src/nodes/mec/MECPlatform/MECServices/LocationService/LocationService.o:(vtable for LocationService)
referenced by ../out/gcc-release/src/nodes/mec/MECPlatform/MECServices/MECServiceBase/MecServiceBase.o:(vtable for MecServiceBase)
referenced by ../out/gcc-release/src/nodes/mec/MECPlatform/MECServices/MECServiceBase/MecServiceBase.o:(vtable for inet::OperationalBase)
referenced 3 more times

ld.lld: error: undefined symbol: inet::OperationalMixin<omnetpp::cSimpleModule>::isDown() const
referenced by ../out/gcc-release/src/nodes/mec/MECPlatform/MECServices/LocationService/LocationService.o:(vtable for LocationService)
referenced by ../out/gcc-release/src/nodes/mec/MECPlatform/MECServices/MECServiceBase/MecServiceBase.o:(vtable for MecServiceBase)
referenced by ../out/gcc-release/src/nodes/mec/MECPlatform/MECServices/MECServiceBase/MecServiceBase.o:(vtable for inet::OperationalBase)
referenced 3 more times

ld.lld: error: undefined symbol: inet::OperationalMixin<omnetpp::cSimpleModule>::numInitStages() const
referenced by ../out/gcc-release/src/nodes/mec/MECPlatform/MECServices/MECServiceBase/MecServiceBase.o:(vtable for inet::OperationalBase)
collect2.exe: error: ld returned 1 exit status
C:/Users/Desktop/omnetpp-6.0.1/tools/win32.x86_64/mingw64/bin/llvm-ar.exe: error: unable to load '../out/gcc-release/src/libsimu5g.dll.a': No such file or directory
make[1]: *** [Makefile:424: ../out/gcc-release/src/libsimu5g.dll] Error 1
make[1]: Leaving directory '/c/Users/Desktop/omnetpp-6.0.1/samples/simu5g/src'
make: *** [Makefile:2: all] Error 2
"make MODE=release all" terminated with exit code 2. Build might be incomplete.
```

**Comment by rhornig** [comment, marked as answer]
I've seen similar issues with other project where the virtual tables for templated base classes were not properly exported/generated by the compiler. While the simu5g website indicates that it supposed to be working on Windows, I'm not sure whether the authors test it in all possible configurations. 

You seem the be using gcc instead of the default clang compiler. That in itself is enough to run into similar problems on Windows. I would first recompile everything with clang (as hopefully that was tested by the authors). If that is still not working, get in touch with the authors.

An other workaround I would recommend is installing OMNeT++ and Simu5G in WSL on Windows 10. I have pretty good experience with installing an Ubuntu WSL image and install the Linux version of omnet there. You will get MUCH faster compile/link times compared to the native windows version and the whole API exporting nighmare is non-existent on Linux.

**Comment by Anamt761** [comment]
the same with HNOCS, have you any idea while doing the task mapping on NoC, where to map the mapping function file on mesh topology 8x8 async?
